lib/github.ml

```diff
@@ -235,6 +250,61 @@ let assert_tag_exists ~dry_run tag =
   if Vcs.tag_exists ~dry_run repo tag then Ok ()
   else R.error_msgf "%s is not a valid tag" tag
 
+(* Ask the user then push the tag. Guess the ssh URI from the dev-repo.
+   This function can abort:
+   - The user answered "N" to pushing the tag
+   - The push failed
+
+   This function does nothing if the tag is already present on the remote and
+   point to the same ref. *)
+let push_tag ~dry_run ~yes ~dev_repo vcs tag =
+  let remote_has_tag_uptodate () =
+    if dry_run then Ok true (* Assume yes when [dry_run] is enabled *)
+    else
+      Vcs.commit_id ~dirty:false ~commit_ish:tag vcs >>= fun local_rev ->
+      Vcs.ls_remote ~dry_run vcs ~kind:`Tag ~filter:tag dev_repo >>= function
+      | [] -> Ok false
+      | (rev, _) :: _ when local_rev = rev -> Ok true
```

Comment:
Indeed we need to look for peeled refs (`tag^{}`).

Dune-release normally makes annotated tags but do you think we should do something in case of tags is created by the user and isn't annotated ?
This function don't work in this case right now but perhaps we should catch this before and say it's an error ?

Reply:
If we want this to work both for annotated and non-annotated tags, we could look for refs instead of peeled refs (`tag` instead of `tag^{}`) and then compare `local_rev` with `Vcs.commit_id ~dirty:false ~commit_ish:rev vcs`:

```suggestion
      | (rev, _) :: _ when local_rev = Vcs.commit_id ~dirty:false ~commit_ish:rev vcs -> Ok true
```

(if you accept this change, it would be good to also go back to using `tag` instead of `tag^{}`)

Reply:
Good idea !
Non-annotated tags still don't work well with the rest of the code (eg. `Pkg.tag` calls `git describe`).
I added a warning message in this case to help diagnostic.

Reply:
Thanks.